com.microsoft.mrtk.input/Simulation/InputSimulator.cs

```diff
@@ -718,6 +719,32 @@ private string GetControlSetName(SimulatorControlScheme set)
             return set.name;
         }
 
+        internal static unsafe bool TryExecuteCommand(InputDeviceCommand* commandPtr, out long result)
+        {
+            // This is a utility method called by XRSimulatedHMD and XRSimulatedController
+            // since both devices share the same command handling.
+            // This replicates the logic in XRToISXDevice::IOCTL (XRInputToISX.cpp).
+            var type = commandPtr->type;
+            if (type == RequestSyncCommand.Type)
+            {
+                // The state is maintained by XRDeviceSimulator, so no need for any change
+                // when focus is regained. Returning success instructs Input System to not
+                // reset the device.
+                result = InputDeviceCommand.GenericSuccess;
+                return true;
+            }
+
+            if (type == QueryCanRunInBackground.Type)
+            {
+                ((QueryCanRunInBackground*)commandPtr)->canRunInBackground = true;
```

Comment:
maybe add a comment here, it seems strange to see c++ syntax in c# code :smile:

Reply:
Commented